mobile/libuplink_android/app/src/androidTest/java/io/storj/mobile/libuplink/LibuplinkInstrumentedTest.java

```diff
@@ -0,0 +1,350 @@
+    public static final String VALID_API_KEY = InstrumentationRegistry.getArguments().getString("api.key", "GBK6TEMIPJQUOVVN99C2QO9USKTU26QB6C4VNM0=");
+
+    String filesDir;
+
+    @Before
+    public void setUp() {
+        filesDir = InstrumentationRegistry.getTargetContext().getFilesDir().getAbsolutePath();
+    }
+
+    @Test
+    public void testOpenProjectFail() throws Exception {
+        Config config = new Config();
+
+        Uplink uplink = new Uplink(config, filesDir);
+        try {
+            ProjectOptions options = new ProjectOptions();
+            options.setEncryptionKey("TestEncryptionKey".getBytes());
+
+            Project project = null;
+            try {
+                // 10.0.2.2 refers to not existing satellite
+                project = uplink.openProject("10.0.2.2:1", VALID_API_KEY, options);
+                fail("exception expected");
+            } catch (Exception e) {
+                // skip
+            } finally {
+                if (project != null) {
+                    project.close();
+                }
+            }
+        } finally {
+            uplink.close();
+        }
+    }
+
+    @Test
+    public void testBasicBucket() throws Exception {
+        Config config = new Config();
+
+        Uplink uplink = new Uplink(config, filesDir);
+        try {
+            ProjectOptions options = new ProjectOptions();
+            options.setEncryptionKey("TestEncryptionKey".getBytes());
+
+            Project project = null;
+            try {
+                project = uplink.openProject(VALID_SATELLITE_ADDRESS, VALID_API_KEY, options);
+
+                String expectedBucket = "testBucket";
+                project.createBucket(expectedBucket, new BucketConfig());
+                BucketInfo bucketInfo = project.getBucketInfo(expectedBucket);
+                Assert.assertEquals(expectedBucket, bucketInfo.getName());
+
+                project.deleteBucket(expectedBucket);
+
+                try {
+                    project.getBucketInfo(expectedBucket);
+                } catch (Exception e) {
+                    Assert.assertTrue(e.getMessage().startsWith("bucket not found"));
+                }
+            } finally {
+                if (project != null) {
+                    project.close();
+                }
+            }
+        } finally {
+            uplink.close();
+        }
+    }
+
+    @Test
+    public void testListBuckets() throws Exception {
+        Config config = new Config();
+
+        Uplink uplink = new Uplink(config, filesDir);
+        try {
+            ProjectOptions options = new ProjectOptions();
+            options.setEncryptionKey("TestEncryptionKey".getBytes());
+
+            Project project = null;
+            try {
+                project = uplink.openProject(VALID_SATELLITE_ADDRESS, VALID_API_KEY, options);
```

Comment:
I think this can be outside the try-finally block. It will also simplify closing in the finally block, i.e.
```java
Project project = uplink.openProject(VALID_SATELLITE_ADDRESS, VALID_API_KEY, options);
try {
    ...
} finally {
    project.close();
}
```